### PR TITLE
Corrections and additions to FileDialog

### PR DESCRIPTION
**Pull request by Ovnuniarchos**
Exposed mode, access, filters, and show hidden files to the editor. Also, two functions to get/set the filter list.

### CODE REVIEW

File: scene/gui/file_dialog.cpp [hunk 1]
```diff
@@ -475,7 +475,7 @@ void FileDialog::update_filters() {
 		String flt=filters[i].get_slice(";",0).strip_edges();
 		String desc=filters[i].get_slice(";",1).strip_edges();
 		if (desc.length())
-			filter->add_item(desc+" ( "+flt+" )");
+			filter->add_item(RTR(desc)+" ( "+flt+" )");
```
Comment (vnen) [inline scene/gui/file_dialog.cpp]: `RTR(desc)` is incorrect, it's not possible to translate a variable (the translatable strings are extracted from the *source code*).

**Comment by Ovnuniarchos** [comment]
Should be correct now.

**Comment by Ovnuniarchos** [comment]
I forgot to add the changes to the commit. I removed an unnecessary \#include.

**Comment by akien-mga** [comment]
Code looks fine. Any opinion about the added feature?

**Pull request merged**